src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorPlugin.cpp

```diff
@@ -0,0 +1,160 @@
+        rsuStatus_t.join();
+    }
+
+    void RSUHealthMonitorPlugin::UpdateConfigSettings()
+    {
+        PLOG(logINFO) << "Updating configuration settings.";
+
+        lock_guard<mutex> lock(_configMutex);
+        GetConfigValue<uint16_t>("Interval", _interval);
+        GetConfigValue<string>("RSUIp", _rsuIp);
+        GetConfigValue<uint16_t>("SNMPPort", _snmpPort);
+        GetConfigValue<string>("AuthPassPhrase", _authPassPhrase);
+        GetConfigValue<string>("SecurityUser", _securityUser);
+        GetConfigValue<string>("SecurityLevel", _securityLevel);
+        GetConfigValue<string>("RSUMIBVersion", _rsuMIBVersionStr);
+        boost::trim_left(_rsuMIBVersionStr);
+        boost::trim_right(_rsuMIBVersionStr);
+        // Support RSU MIB version 4.1
+        if (boost::iequals(_rsuMIBVersionStr, RSU4_1_str))
+        {
+            _rsuMibVersion = RSUMIB_V_4_1;
+        }
+        else
+        {
+            _rsuMibVersion = UNKOWN_MIB_V;
+            PLOG(logERROR) << "Uknown RSU MIB version: " <<_rsuMIBVersionStr;
+        }
+    }
+
+    void RSUHealthMonitorPlugin::OnConfigChanged(const char *key, const char *value)
+    {
+        PluginClient::OnConfigChanged(key, value);
+        UpdateConfigSettings();
+    }
+
+    void RSUHealthMonitorPlugin::PeriodicRSUStatusReq()
+    {
+        while (true)
+        {
+            try
+            {
+                // Periodic SNMP call to get RSU status based on RSU MIB version 4.1
+                auto rsuStatusJson = getRSUStatus();
+                // Broadcast the RSU status info when there are RSU responses.
+                if (!rsuStatusJson.empty())
+                {
+                    vector<string> rsuStatusFields;
+                    for (auto const &field : rsuStatusJson.getMemberNames())
+                    {
+                        rsuStatusFields.push_back(field);
+                    }
+                    // Only broadcast RSU status when all required fields are present.
+                    if (_rsuWorker && _rsuWorker->isAllRequiredFieldsPresent(_rsuMibVersion, rsuStatusFields))
+                    {
+                        Json::FastWriter fasterWirter;
+                        string json_str = fasterWirter.write(rsuStatusJson);
+                        tmx::messages::RSUStatusMessage sendRsuStatusMsg;
+                        sendRsuStatusMsg.set_contents(json_str);
+                        string source = RSUHealthMonitorPlugin::GetName();
+                        BroadcastMessage(sendRsuStatusMsg, source);
+                        PLOG(logINFO) << "Broadcast RSU status:  " << json_str;
+                    }
+                }
+            }
+            catch (const std::exception &ex)
+            {
+                PLOG(logERROR) << ex.what();
+            }
+            this_thread::sleep_for(chrono::seconds(_interval));
+        }
+    }
+
+    Json::Value RSUHealthMonitorPlugin::getRSUStatus()
```

Comment:
Is there are reason this is not in the RSU Worker. Here it is not unit testable?

Reply:
It is sending the SNMP request which requires the connection to the RSU, and also require the global RSU Status config variable from the plugin.

Reply:
Both the SNMP Client and any configuration can be passed to another class to make this call right? Just my opinion, but when creating new V2X-Hub Plugins since we can not unit test the code there, we should be moving most of it to other cpp files. Also for maintainability reasons we should not couple the logic for making snmp request, json serialization and the plugin core logic together if we do not need to.